server/index.js

```diff
@@ -29,6 +29,34 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+  const sql = `
+    select *
+    from "products"
+    where "productId" = $1
+  `;
+
+  const params = [productId];
+  if (productId > 6 || productId <= 0) {
+    return next(new ClientError('Invalid product', 400));
+  } else if (productId === '') {
+    res.status(500).json({
+      error: 'An unexpected error occurred'
+    });
```

Comment:
You can leave this check out. If the request matches `/api/products/:productId`, you're guaranteed to have _something_ inside `req.params.productId`.